crates/client/rpc/src/lib.rs

```diff
@@ -843,7 +849,11 @@ where
             StarknetRpcApiError::BlockNotFound
         })?;
 
-        let starknet_block = get_block_by_block_hash(self.client.as_ref(), substrate_block_hash).unwrap_or_default();
+        let starknet_block = get_block_by_block_hash(self.client.as_ref(), substrate_block_hash).ok_or_else(|| {
+            error!("Block not found");
```

Comment:
we can move the "Block not found" log inside `get_block_by_block_hash` and throw the error directly here after implementing what @tdelabro said.